### PR TITLE
Added image loading in SUNRGB-D dataset

### PR DESCRIPTION
**Pull request by THU17cyz**
Also fixed a bug in sunrgbd_data_utils.py; created a testing imvotenet config file to test that the loading is correct

### CODE REVIEW

File: configs/imvotenet/imvotenet_temp.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        test_mode=True,
+        box_type_3d='Depth'),
+    test=dict(
+        type=dataset_type,
+        data_root=data_root,
+        ann_file=data_root + 'sunrgbd_infos_val.pkl',
+        pipeline=test_pipeline,
+        classes=class_names,
+        test_mode=True,
+        box_type_3d='Depth'))
```
Comment (Tai-Wang) [inline configs/imvotenet/imvotenet_temp.py:117]: Rename the config like other model configs

File: mmdet3d/models/detectors/imvotenet.py [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+from mmdet3d.core import bbox3d2result, merge_aug_bboxes_3d
+from mmdet.models import DETECTORS
+from .single_stage import SingleStage3DDetector
+
+
+@DETECTORS.register_module()
+class ImVoteNet(SingleStage3DDetector):
+    """ImVoteNet model.
+
+    https://arxiv.org/pdf/2001.10692.pdf
```
Comment (Tai-Wang) [inline mmdet3d/models/detectors/imvotenet.py:12]: r"""`ImVoteNet <https://arxiv.org/pdf/2001.10692.pdf>`_ for 3D detection."""
I find the docstring in VoteNet also has the similar problem. Please also modify it accordingly.

File: mmdet3d/models/detectors/imvotenet.py [hunk 3]
```diff
@@ -0,0 +1,132 @@
...
+            backbone=backbone,
+            bbox_head=bbox_head,
+            train_cfg=train_cfg,
+            test_cfg=test_cfg,
+            pretrained=pretrained)
+
+    def extract_img_feat(self, img, img_metas):
+        """Extract features of images."""
+        print(img, img_metas)  # test
+        assert False
```
Comment (Tai-Wang) [inline mmdet3d/models/detectors/imvotenet.py:31]: line 30 and 31 can be removed?

File: mmdet3d/models/detectors/imvotenet.py (hunk 2 above)
Comment (THU17cyz) [reply mmdet3d/models/detectors/imvotenet.py]: I find that many other models do not follow this format

Comment (THU17cyz) [reply mmdet3d/models/detectors/imvotenet.py]: Shall I fix them too?

**Comment by THU17cyz** [comment]
removed imvotenet relevant files

File: mmdet3d/models/detectors/votenet.py [hunk 4]
```diff
@@ -7,10 +7,7 @@
 
 @DETECTORS.register_module()
 class VoteNet(SingleStage3DDetector):
-    """VoteNet model.
-
-    https://arxiv.org/pdf/1904.09664.pdf
-    """
+    r"""VoteNet `<https://arxiv.org/pdf/1904.09664.pdf>`_ for 3D detection."""
```
Comment (ZwwWayne) [inline mmdet3d/models/detectors/votenet.py:10]:
```python
VoteNet `<https://arxiv.org/pdf/1904.09664.pdf>`_ -> `VoteNet <https://arxiv.org/pdf/1904.09664.pdf>`_
```

File: mmdet3d/datasets/sunrgbd_dataset.py [hunk 5]
```diff
@@ -60,6 +60,43 @@ def __init__(self,
...
+                - ann_info (dict): Annotation info.
+        """
+        info = self.data_infos[index]
+        assert info['point_cloud']['lidar_idx'] == info['image']['image_idx']
+        sample_idx = info['point_cloud']['lidar_idx']
+        pts_filename = osp.join(self.data_root, info['pts_path'])
+        img_filename = osp.join(self.data_root, info['image']['image_path'])
+        calib = info['calib']
+
+        input_dict = dict(
```
Comment (ZwwWayne) [inline mmdet3d/datasets/sunrgbd_dataset.py:85]: We should support to decide which modality data to be load according to `self.modality`.

Comment (THU17cyz) [reply mmdet3d/datasets/sunrgbd_dataset.py:84]: By the way, I noticed that self.modality in kitti_dataset.py isn't used

**Comment by filaPro** [comment]
Hi @THU17cyz,

(May be not the best place for this question, but anyway) Do you have plans to finish ImVoteNet implemetation? Thank you.

**Comment by THU17cyz** [comment]
> Hi @THU17cyz,
> 
> (May be not the best place for this question, but anyway) Do you have plans to finish ImVoteNet implemetation? Thank you.

Yes, I'm working on reproducing ImVoteNet. But since I'm not working full-time, I may take some time.

File: mmdet3d/datasets/sunrgbd_dataset.py [hunk 6]
```diff
@@ -46,7 +46,7 @@ def __init__(self,
                  ann_file,
                  pipeline=None,
                  classes=None,
-                 modality=None,
+                 modality='pconly',
```
Comment (ZwwWayne) [inline mmdet3d/datasets/sunrgbd_dataset.py:49]: Use a similar protocol as those in nuscenes dataset.
```python
input_modality = dict(use_lidar=True, use_camera=True)
```

**Review by Tai-Wang — COMMENTED** [review]
Just need to modify according to current comments. No additional problem from my point of view.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**